### PR TITLE
Improve styling of confirmation PDFS

### LINKED ISSUE #949: Restyle the confirmation PDF to look similar to the confirmation email
The confirmation PDF could re-use alot of styling and logic from the email (specifically the summary tag but then for all fields).

**Tasks**

* [x] Restyle PDF - should look more like the confirmation page in the SDK
    * Group data per form step
    * Get rid of ugly table borders etc, visual identity should be consistent
* [x] Add text shown on the confirmation page, at the bottom of the PDF.

### PR DESCRIPTION
**Pull request by sergei-maertens**
Fixes #949 

* Styling matches SDK confirmation page & confirmation e-mail
* Header with background color from design tokens respected
* If a logo is uploaded, it is included in the header as well
* No footer - has no added value

Preview: 
image

### CODE REVIEW

**Review by SilviaAmAm — COMMENTED** [review]
It looks good! I have a question though. When I look at the PDF through the /dev endpoint, i see it correctly formatted:
Screenshot from 2022-03-04 17-13-40

But when I download it after completing a form I see this:
Screenshot from 2022-03-04 17-14-25

In celery I see these errors:
```
[2022-03-04 17:09:52,436: ERROR/ForkPoolWorker-1] Relative URI reference without a base URI: <link href="/static/bundles/pdf-css.css">
[2022-03-04 17:09:52,439: ERROR/ForkPoolWorker-1] Relative URI reference without a base URI: <img src="/media/logo/gemeente_logo_duiven-460x200.jpg">
```

Maybe I configured something wrong?

File: src/openforms/submissions/templates/report/submission_report.html [hunk 1]
```diff
@@ -1,62 +1,86 @@
...
-            width: 100%;
-        }
-        .table-cell-key {
-            font-weight: bold;
-        }
-        .contact-details, .help {
-            background: black;
-            color: white;
+{% load static i18n solo_tags style_dictionary %}<!DOCTYPE html>
+{# https://github.com/sergei-maertens/nyvellum/blob/master/src/nyvellum/contrib/bolcom/templates/admin/bolcom/shipment/invoice.html for reference #}
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/submissions/templates/report/submission_report.html]: yeah, I removed the comment as it was a self-reminder :grimacing:

File: src/openforms/submissions/dev_views.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    .. todo:: refactor view/method to always use same template/context without
+       duplicating code
+    """
+
+    template_name = "report/submission_report.html"
+    queryset = Submission.objects.all()
+    context_object_name = "submission"
+
+    def dispatch(self, request, *args, **kwargs):
+        if not settings.DEBUG or not request.user.is_superuser:
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/submissions/dev_views.py:27]: I've added those tests, including the e-mail wrapper as well

**Comment by sergei-maertens** [comment]
Fixed some more WeasyPrint related bugs/mistakes in the `UrlFetcher` implementation, but at least in dev settings we can now see the log output from `WeasyPrint` and there's a dev-only management command to test it easily.

**Pull request merged**